packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -260,6 +312,32 @@ impl Abigen {
     pub fn get_types(abi: &ProgramABI) -> HashMap<usize, TypeDeclaration> {
         abi.types.iter().map(|t| (t.type_id, t.clone())).collect()
     }
+
+    pub fn resolve_logs(&self) -> Result<Vec<TokenStream>, Error> {
+        let resolved = self
+            .abi
+            .logged_types
+            .as_ref()
+            .map_or(vec![], |logged_types| {
+                logged_types
+                    .iter()
+                    .map(|l| {
+                        let resolved_type = resolve_type(&l.application, &self.types)
+                            .expect("Failed to resolve log type");
+                        let param_type_call = single_param_type_call(&resolved_type);
+                        let id = l.log_id;
+
+                        quote! { ResolvedLog {
+                                log_id: #id,
+                                param_type: #param_type_call
+                            }
+                        }
+                    })
+                    .collect()
+            });
```

Comment:
I find `flattening` options of containers more pleasable than handling the collection separately, YMMV:
```rust
        self.abi
            .logged_types
            .as_ref()
            .into_iter()
            .flatten()
            .map(|logged_type| {
                let resolved_type = resolve_type(&logged_type.application, &self.types)
                    .expect("Failed to resolve log type");
                let param_type_call = single_param_type_call(&resolved_type);
                let id = logged_type.log_id;

                quote! { ResolvedLog {
                        log_id: #id,
                        param_type: #param_type_call
                    }
                }
            })
            .collect()
```